rfcs/0211-close-transactions-on-daily-closing.md

```diff
@@ -0,0 +1,100 @@
+>    ```json
+>    {
+>      "CurrentStartedTransactionNumbers": [ 10, 12, 13 ]
+>    }
+>    ```
+> 2. A Daily-Closing Request with the ftReceiptCase `0x4445000300000007` (Daily-Closing with Close Open Transactions flag `0x200000000`) is sent.
+>    The open transactions are closed by the middleware.
+> 3. A Zero-Receipt Request with the ftReceiptCase `0x4445000100800002` is sent again in the Response there are no open transactions on the TSE.
+>    ```json
+>    {
+>      "CurrentStartedTransactionNumbers": []
+>    }
+>    ```
+
+# Reference-level explanation
+
+The Daily-Closing already gets the TSE-Info from the TSE and the middleware already has the ability to close open transactions on the TSE as done in the Fail-Transaction Receipt.
+
+In the we now introduce a check for the new flag and close the transactions if it is set.
+
+```cs
+if (request.HasCloseOpenTransactionsOnTseFlag() && tseInfo.CurrentStartedTransactionNumbers!.Any())
+{
+    await CloseOpenTransactionsOnTseAsync(tseInfo.CurrentStartedTransactionNumbers);
+}
+```
+
+The method `CloseOpenTransactionsOnTseAsync` looks like this which is inspired by the current [Fail-Transaction Receipt](https://github.com/fiskaltrust/middleware/blob/1a9abd80430e9dfecdd17289024e9d19e798d19b/queue/src/fiskaltrust.Middleware.Localization.QueueDE/RequestCommands/FailTransactionReceiptCommand.cs#L64-L77).
+
+```cs
+private CloseOpenTransactionsOnTseAsync(IEnumerable<long> currentStartedTransactionNumbers)
+  var openSignatures = new List<SignaturItem>();
+  var openTransactions = (await _openTransactionRepo.GetAsync().ConfigureAwait(false)).ToList();
+  var transactionsToClose = JsonConvert.DeserializeObject<TseInfo>(request.ftReceiptCaseData);
+  foreach (var openTransactionNumber in currentStartedTransactionNumbers)
+  {
+      (var openProcessType, var openPayload) = _transactionPayloadFactory.CreateAutomaticallyCanceledReceiptPayload();
+      var finishResult = await _transactionFactory.PerformFinishTransactionRequestAsync(openProcessType, openPayload, queueItem.ftQueueItemId, queueDE.CashBoxIdentification, openTransactionNumber).ConfigureAwait(false);
+      openSignatures.AddRange(_signatureFactory.GetSignaturesForFinishTransaction(finishResult));
+      var openTransaction = openTransactions.FirstOrDefault(x => (ulong) x.TransactionNumber == openTransactionNumber);
+      if (openTransaction != null)
+      {
+          await _openTransactionRepo.RemoveAsync(openTransaction.cbReceiptReference).ConfigureAwait(false);
+      }
+  }
+}
+```
+
+# Drawbacks
+
+## Latency
+
+If there are hundreds of open transaction on the TSE closing them all can take a while.
+This could lead to a timeout or long waiting times when performing a Daily-Closing.
+
+## Closing foreign Transactions
+
+If a TSE is used by multiple Queues the Transactions of open other Queues will be closed as well.
```

Comment:
Might make sense to protocol in the actionjournal which transactions were closed 🤔 in one entry, not X :D

Reply:
Good idea. I've added pseudo code for that 🙏 I'm not sure though about the ActionJournal Type though